pkg/registry/registry.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/k8s/client"
+	"github.com/okteto/okteto/pkg/k8s/namespaces"
+	"github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+)
+
+//GetImageTagWithDigest returns the image tag diggest
+func GetImageTagWithDigest(ctx context.Context, imageTag string) (string, error) {
+	registryURL, err := okteto.GetRegistry()
+	if err != nil {
+		log.Infof("error accessing to okteto registry: %s", err.Error())
+		return imageTag, nil
+	}
+	expandedTag, err := ExpandOktetoDevRegistry(ctx, imageTag)
+	if err != nil {
+		log.Infof("error expanding okteto registry: %s", err.Error())
+		return imageTag, nil
+	}
+	if !strings.HasPrefix(expandedTag, registryURL) {
+		return imageTag, nil
+	}
+	username := okteto.GetUserID()
+	token, err := okteto.GetToken()
```

Comment:
is the GetImageTagWithDigest function only called for okteto managed images and registry?

Reply:
no, but it would exit here:
https://github.com/okteto/okteto/pull/1154/files#diff-cb61d8f79451b9541de4a8cc0811523a68d15452b2f5971c7618ea5b423cf4ecR40